src/libraries/Microsoft.Extensions.Options/tests/Microsoft.Extensions.Options.Tests/OptionsBuilderValidationTests.cs

```diff
@@ -0,0 +1,453 @@
+            {
+                services.AddOptions<ComplexOptions>()
+                    .Configure(o => o.Boolean = false)
+                    .Validate(o => o.Boolean, "Boolean must be true.")
+                    .ValidateOnStart();
+            });
+
+            using (var host = hostBuilder.Build())
+            {
+                var error = await Assert.ThrowsAsync<OptionsValidationException>(async () =>
+                {
+                    await host.StartAsync();
+                });
+
+                ValidateFailure<ComplexOptions>(error, 1, "Boolean must be true.");
+            }
+        }
+
+        internal class FakeService { }
+
+        internal class FakeSettings
+        {
+            public string Name { get; set; }
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34580", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        public async Task ValidateOnStart_NamedOptions_ValidatesFailureOnStart()
+        {
+            var hostBuilder = CreateHostBuilder(services =>
+            {
+                services.AddOptions().AddSingleton(new FakeService());
+                services
+                    .AddOptions<FakeSettings>("named")
+                    .Configure<FakeService>((o, _) =>
+                    {
+                        o.Name = "named";
+                    })
+                    .Validate(o => o.Name == null, "trigger validation failure for named option!")
+                    .ValidateOnStart();
+            });
+
+            using (var host = hostBuilder.Build())
+            {
+                var error = await Assert.ThrowsAsync<OptionsValidationException>(async () =>
+                {
+                    await host.StartAsync();
+                });
+
+                ValidateFailure<FakeSettings>(error, 1, "trigger validation failure for named option!");
+            }
+        }
+
+        [Fact]
+        [ActiveIssue("https://github.com/dotnet/runtime/issues/34580", TestPlatforms.Windows, TargetFrameworkMonikers.Netcoreapp, TestRuntimes.Mono)]
+        private async Task ValidateOnStart_AddOptionsMultipleTimesForSameType_LastValidateOnStartCallGetsTriggered()
+        {
+            var hostBuilder = CreateHostBuilder(services =>
+            {
+                services.AddOptions<ComplexOptions>("bad_configuration1")
+                    .Configure(o => o.Boolean = false)
+                    .Validate(o => o.Boolean, "bad_configuration1")
+                    .ValidateOnStart();
+
+                services.AddOptions<ComplexOptions>("bad_configuration2")
+                    .Configure(o =>
+                    {
+                        o.Boolean = false;
+                        o.Integer = 11;
+                    })
+                    .Validate(o => o.Boolean, "Boolean")
+                    .Validate(o => o.Integer > 12, "Integer")
+                    .ValidateOnStart();
```

Comment:
Because of

https://github.com/dotnet/runtime/blob/68e1b66345d8e4d8174f61c892201f76a18331b9/src/libraries/Microsoft.Extensions.Options/src/OptionsBuilderValidationExtensions.cs#L34

If we modified `CreateHostBuilder` in this test to have a correct configuration as its last ValidateOnStart options setup, then we get no failures on startup at all. (check below) 

```c#


            var hostBuilder = CreateHostBuilder(services =>
            {
                services.AddOptions<ComplexOptions>("bad_configuration1")
                    .Configure(o => o.Boolean = false)
                    .Validate(o => o.Boolean, "bad_configuration1")
                    .ValidateOnStart();

                services.AddOptions<ComplexOptions>("bad_configuration2")
                    .Configure(o =>
                    {
                        o.Boolean = false;
                        o.Integer = 11;
                    })
                    .Validate(o => o.Boolean, "Boolean")
                    .Validate(o => o.Integer > 12, "Integer")
                    .ValidateOnStart();

                services.AddOptions<ComplexOptions>("correct_configuration")
                    .Configure(o => o.Boolean = true)
                    .Validate(o => o.Boolean, "Boolean must be true.")
                    .ValidateOnStart();
            });
```

Reply:
That's by design. Validation runs after constructing the final object by running all of the delegates. Validation is attached to the TOptions type not the specific call to AddOptions